Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/ComponentEditor.cpp

```diff
@@ -144,23 +145,19 @@ namespace AzToolsFramework
     ComponentEditor::ComponentEditor(
         AZ::SerializeContext* context,
         IPropertyEditorNotify* notifyTarget /* = nullptr */,
-        QWidget* parent /* = nullptr */,
-        bool replaceRPE /* = false */,
-        AZStd::shared_ptr<AZ::DocumentPropertyEditor::ComponentAdapter> customDpeComponentAdapter /* = nullptr */)
+        QWidget* parent /* = nullptr */)
         : AzQtComponents::Card(new ComponentEditorHeader(), parent)
         , m_serializeContext(context)
     {
         GetHeader()->SetTitle(ComponentEditorConstants::kUnknownComponentTitle);
 
-        if (replaceRPE)
+        if (DocumentPropertyEditor::ShouldReplaceRPE())
         {
             // Instantiate the DPE without the RPE
-            m_adapter = customDpeComponentAdapter;
-            if (!m_adapter)
-            {
-                // Create a default component adapter.
-                m_adapter = AZStd::make_shared<AZ::DocumentPropertyEditor::ComponentAdapter>();
-            }
+            m_adapter = (Prefab::IsInspectorOverrideManagementEnabled() ?
+                AZStd::make_shared<Prefab::PrefabComponentAdapter>() :
+                AZStd::make_shared<AZ::DocumentPropertyEditor::ComponentAdapter>());
+
```

Comment:
ugh. Any idea why those tools don't work with the DPE? I'd be surprised if they have any special requirements that the Inspector doesn't...

Reply:
I think they don't work because the ui editor still uses slices and therefore we shouldn't create a prefab adapter for those editors. So, when IsInspectorOverrideManagementEnabled() is true or when the feature is launched, this will stop working for ui editor components.

Reply:
Is there a different flag we can be checking here, then? It's really nasty to have to pass replaceRPE *and* a customDpeComponentAdapter. Partly because it's redundant, but also because multi-edit would somehow have to create more instances of whatever that customDpeComponentAdapter as aggregate instances are added.

Reply:
If we want to restrict adapter type in ComponentEditor, we could just pass the "enableDPE" flag in the constructor.

Reply:
It would mean that other tools using ComponentEditor (afaik UI Editor and Script Canvas) would choose to enable DPE or not, but if DPE is enabled and overrides are enabled then the tool would always be using the prefab component adapter which may or may not be desirable.

Reply:
Another option could be to pass in an "adapter creation" callback to the ComponentEditor constructor. Then DPE enabling could be determined if the callback is set.

The original idea with the custom adapter was to allow ComponentEditor to be used by non-DPE enabled and/or non-Prefab enabled tools, and our DPE/Prefab enabled flags span across all tools.

Reply:
yep! I went with the "adapter creation" factory, thanks!